pkg/scaling/scale_handler.go

```diff
@@ -611,3 +607,152 @@ func (h *scaleHandler) getScaledObjectState(ctx context.Context, scaledObject *k
+			if m.MetricName == metricSpecs[0].External.Metric.Name {
+				metricValue = m.Value.AsApproximateFloat64()
+				queueLength += metricValue
+			}
+		}
+		scalerLogger.V(1).Info("Scaler Metric value", "isTriggerActive", isTriggerActive, metricSpecs[0].External.Metric.Name, queueLength, "targetAverageValue", targetAverageValue)
+
+		if isTriggerActive {
+			isActive = true
+		}
+
+		if targetAverageValue != 0 {
+			averageLength := queueLength / targetAverageValue
+			maxValue = min(float64(scaledJob.MaxReplicaCount()), averageLength)
+		}
+		scalersMetrics = append(scalersMetrics, scalerMetrics{
+			queueLength: queueLength,
+			maxValue:    maxValue,
+			isActive:    isActive,
+		})
+	}
+	return scalersMetrics
+}
+
+// isScaledJobActive returns whether the input ScaledJob:
+// is active as the first return value,
+// the second and the third return values indicate queueLength and maxValue for scale
+func (h *scaleHandler) isScaledJobActive(ctx context.Context, scaledJob *kedav1alpha1.ScaledJob) (bool, int64, int64) {
+	var queueLength float64
+	var maxValue float64
+	isActive := false
+
+	logger := logf.Log.WithName("scalemetrics")
+	scalersMetrics := h.getScaledJobMetrics(ctx, scaledJob)
+	switch scaledJob.Spec.ScalingStrategy.MultipleScalersCalculation {
+	case "min":
+		for _, metrics := range scalersMetrics {
+			if (queueLength == 0 || metrics.queueLength < queueLength) && metrics.isActive {
+				queueLength = metrics.queueLength
+				maxValue = metrics.maxValue
+				isActive = metrics.isActive
+			}
+		}
+	case "avg":
+		queueLengthSum := float64(0)
+		maxValueSum := float64(0)
+		length := 0
+		for _, metrics := range scalersMetrics {
+			if metrics.isActive {
+				queueLengthSum += metrics.queueLength
+				maxValueSum += metrics.maxValue
+				isActive = metrics.isActive
+				length++
+			}
+		}
+		if length != 0 {
+			queueLength = queueLengthSum / float64(length)
+			maxValue = maxValueSum / float64(length)
+		}
+	case "sum":
+		for _, metrics := range scalersMetrics {
+			if metrics.isActive {
+				queueLength += metrics.queueLength
+				maxValue += metrics.maxValue
+				isActive = metrics.isActive
+			}
+		}
+	default: // max
+		for _, metrics := range scalersMetrics {
+			if metrics.queueLength > queueLength && metrics.isActive {
+				queueLength = metrics.queueLength
+				maxValue = metrics.maxValue
+				isActive = metrics.isActive
+			}
+		}
+	}
+
+	if scaledJob.MinReplicaCount() > 0 {
+		isActive = true
+	}
+
+	maxValue = min(float64(scaledJob.MaxReplicaCount()), maxValue)
+	logger.V(1).WithValues("ScaledJob", scaledJob.Name).Info("Checking if ScaleJob Scalers are active", "isActive", isActive, "maxValue", maxValue, "MultipleScalersCalculation", scaledJob.Spec.ScalingStrategy.MultipleScalersCalculation)
+
+	return isActive, ceilToInt64(queueLength), ceilToInt64(maxValue)
+}
+
+func ceilToInt64(x float64) int64 {
+	return int64(math.Ceil(x))
+}
+
+// Min function for float64
+func min(x, y float64) float64 {
```

Comment:
Could you please move this to a separate function in a package pkg/scaling/scaledjob? And refactor the code a little bit so these don't have to be a public functions if possible.

Reply:
Done.